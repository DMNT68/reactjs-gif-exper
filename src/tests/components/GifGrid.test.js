import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba en <GifGrid />', () => {
  const category = 'one punch';

  test('debe mostar el componente correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imagenes useGetchGifs', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Cualquie cosa',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  });
});
