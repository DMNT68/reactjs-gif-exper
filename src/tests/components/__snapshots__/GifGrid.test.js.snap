// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGrid /> debe de mostrar items cuando se cargan imagenes useGetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    one punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquie cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Prueba en <GifGrid /> debe mostar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    one punch
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
